### ISSUE TITLE
App crashes on quitting in macOS

### ISSUE THREAD

**Issue by chaitanyaprem** [status-im/status-desktop#13588]
# Bug Report
Noticed many times that app crashes after quitting in macOS. 

## Description

Attaching crash report and app logs if it helps.

## Steps to reproduce

1. Right-click on app icon in toolbar and select `Quit`
2. app crashes and generates a crash report (attaching a sample report)

[crash_report.txt](https://github.com/status-im/status-desktop/files/14306001/crash_report.txt)
[app_20240216_101307.log](https://github.com/status-im/status-desktop/files/14306008/app_20240216_101307.log)

#### Expected behavior
App should not crash on quitting.

### Additional Information

- Status desktop version: Version: 0.9.90, Commit: 7b302c07e
- Operating System: Sonoma 14.1.1 (23B81)

**Comment by chaitanyaprem** [comment]
I tried it today and it crashed again in the release version.
Hope you have followed the steps mentioned above. 
Also, it could be due to some race conditions as well, and maybe not reproducible all the times. 

Here is the crash report. 
[crash_report_status.log](https://github.com/status-im/status-desktop/files/15441158/crash_report_status.log)

**Comment by alexjba** [comment]
@mprakhov Only lldb + debug environment. But if it's a qml bug/crash it won't really help..

**Comment by caybro** [comment]
> @mprakhov Only lldb + debug environment. But if it's a qml bug/crash it won't really help..

Yeah, a decent debugger should be good. It would also help a bit more if you managed to somehow install the missing symbols

> frame #1: 0x000000010431eefc QtQuick`___lldb_unnamed_symbol13718 + 60

but otherwise what @alexjba said
